### PR TITLE
36 settingsscreen

### LINKED ISSUE #36: Settings screen
A screen where things like appearance and soundtrack can be changed

### PR DESCRIPTION
**Pull request by MKerkhof**
Will close #36

### CODE REVIEW

**Comment by MathiasMeuleman** [comment]
At my machine all audioplayer tests and the E2E fail. Any ideas? Rest of the code looks fine!

<feature, review, testing

File: src/main/java/nl/tudelft/ti2206/group9/gui/GameScreen.java [hunk 1]
```diff
@@ -93,6 +100,10 @@ public static void start(final Stage stage) {
 		primaryStage.setResizable(false);
 		primaryStage.show();
 
+        if(SettingsScreen.sound) {
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/GameScreen.java]: Missing space after 'if'.

File: src/main/java/nl/tudelft/ti2206/group9/gui/GameScreen.java [hunk 2]
```diff
@@ -152,6 +163,8 @@ public static void startTickers() {
 
 	/** Resumes the tickers. */
 	public static void resumeTickers() {
+        if (SettingsScreen.sound)
+		audioPlayer.play();
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/GameScreen.java]: if-construct must use brackets { }.

<layout

File: src/main/java/nl/tudelft/ti2206/group9/gui/GameScreen.java [hunk 3]
```diff
@@ -56,9 +58,14 @@
 	/** The primarystage. */
 	private static Stage primaryStage;
 
+	/** The AudioPlayer to be used for background music. */
+    static AudioPlayer audioPlayer = new AudioPlayer("src/main/resources"
+            + "/nl.tudelft.ti2206.group9.audio/soundtrack.aiff");
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/GameScreen.java]: Can I ask why the audioPlayer is not private? :)

File: src/main/java/nl/tudelft/ti2206/group9/gui/SettingsScreen.java [hunk 4]
```diff
@@ -21,6 +22,8 @@
 
 	/** Hide public constructor. */
 	private SettingsScreen() { }
+    /** Boolean for sound status. */
+    public static boolean sound = true;
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/SettingsScreen.java]: This boolean should be togglable with a toggle method, for example:

    public static final boolean isSoundEnabled() {
        return sound;
    }
    public static final void toggleSound() {
        sound = !sound;
    }
    public static final void toggleSound(boolean newState) {
        sound = newState;
    }

File: src/main/java/nl/tudelft/ti2206/group9/audio/AudioPlayer.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	public final String getPath() {
+		return path;
+	}
+
+	/**
+	 * Sets the path of the AudioPlayer.
+	 * @param location new path to a file.
+	 */
+	public final void setPath(final String location) {
+		AudioPlayer.path = location;
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/audio/AudioPlayer.java]: What happens when the path is set to a different path? Shouldn't `initializeTune(path)` be called after setting the path?

<logic

Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/audio/AudioPlayer.java]: This is true, the AudioPlayer should indeed be "reinitialised". 
Will change.
<logic, bug

**Comment by MKerkhof** [comment]
@mpsijm I will fix all the things you mentioned but first I'm waiting for the audioPlayerTest to be fixed, without it, builds will keep on failing.

<bug, fix

**Comment by mpsijm** [comment]
This branch is too old to maintain, and the only real difference with #30 and #42 is that the settings button toggles the sound. This button is already available in master, so the audio branches should be able to add one extra line with `audioplayer.play()` and `.stop()`.

**Pull request closed without merging**